nodes/src/nodes/tool_pipe/IInstance.py

```
@@ -0,0 +1,189 @@
+        data = args.get('data')
+        if not data:
+            raise ValueError('tool_pipe: tool requires a non-empty `data` parameter')
+
+        debug(f'tool_pipe: run_pipe invoked data_len={len(data)}')
+
+        entry = getObject()
+        try:
+            self.instance.open(entry)
+            self._send_to_connected_lane(data)
+            self.instance.close()
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)